bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -0,0 +1,498 @@
+
+import static org.openhab.binding.modbus.sunspec.internal.SunSpecConstants.PROPERTY_UNIQUE_ADDRESS;
+
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.modbus.handler.EndpointNotInitializedException;
+import org.openhab.binding.modbus.handler.ModbusEndpointThingHandler;
+import org.openhab.binding.modbus.sunspec.internal.SunSpecConfiguration;
+import org.openhab.binding.modbus.sunspec.internal.dto.ModelBlock;
+import org.openhab.io.transport.modbus.BasicModbusReadRequestBlueprint;
+import org.openhab.io.transport.modbus.BasicPollTaskImpl;
+import org.openhab.io.transport.modbus.BitArray;
+import org.openhab.io.transport.modbus.ModbusManager;
+import org.openhab.io.transport.modbus.ModbusReadCallback;
+import org.openhab.io.transport.modbus.ModbusReadFunctionCode;
+import org.openhab.io.transport.modbus.ModbusReadRequestBlueprint;
+import org.openhab.io.transport.modbus.ModbusRegisterArray;
+import org.openhab.io.transport.modbus.PollTask;
+import org.openhab.io.transport.modbus.endpoint.ModbusSlaveEndpoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractSunSpecHandler} is the base class for any sunspec handlers
+ * Common things are handled here:
+ *
+ * - loads the configuration either from the configuration file or
+ * from the properties that have been set by the auto discovery
+ * - sets up a regular poller to the device
+ * - handles incoming messages from the device:
+ * - common properties are parsed and published
+ * - other values are submitted to child implementations
+ * - handles disposal of the device by removing any handlers
+ * - implements some tool methods
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractSunSpecHandler extends BaseThingHandler {
+
+    /**
+     * Logger instance
+     */
+    private final Logger logger = LoggerFactory.getLogger(AbstractSunSpecHandler.class);
+
+    /**
+     * Configuration instance
+     */
+    @Nullable
+    protected SunSpecConfiguration config = null;
```

Comment:
```suggestion
    protected @Nullable SunSpecConfiguration config = null;
```

also in other places. And if not needed for a special reason, you can remove the comments above fields. usually the field name is self-explaining its purpose.

Reply:
Ok, changed how @Nullable is written.
Regarding the comments: a few more words in the comments might help someone in the future IMHO.